kernel/src/process/process/mod.rs

```diff
@@ -134,6 +138,11 @@ impl Process {
 
         let prof_clock = ProfClock::new();
 
+        let ppid = match parent.upgrade() {
+            None => 0,
+            Some(parent) => parent.pid(),
```

Comment:
But what if the parent process exits? The cached ppid needs to be kept in sync with the up-to-date parent process.

I think the following structure should be introduced to keep the cached value and the parent process in sync.

```rust
/// Representing a parent process by holding a weak reference to it and its PID.
///
/// This type caches the value of the PID so that it can be retrieved cheaply.
///
/// The benefit of using `ParentProcess` over `(Mutex<Weak<Process>>, Atomic<Pid>,)` is to
/// enforce the invariant that the cached PID and the weak reference are always kept in sync.
pub struct ParentProcess {
    process: Mutex<Weak<Process>>,
    pid: Atomic<Pid>,
}

impl ParentProcess {
    pub fn pid(&self) -> Pid { ... }

    pub fn lock(&mut self) -> ParentProcessGuard<'a> { ... }
}

pub struct ParentProcessGuard<'a> {
    guard: MutexGuard<'a, Weak<Process>>,
    this: &'a ParentProcess,
}

impl ParentProcessGuard {
    pub fn process(&self) -> Weak<Process> { ... }

    pub fn pid(&self) -> Pid { ... }

    pub fn set_process(&mut self, new_process: &Arc<Process>) {
        todo!("update both pid and weak ref");
    }
}
```